consensus/src/experimental/commit_phase.rs

```diff
@@ -385,17 +433,23 @@ impl CommitPhase {
                 );
             }
 
-            if let Some((blocks, ordered_ledger_info, callback)) =
-                self.commit_channel_recv.next().await
-            {
-                report_err!(
-                    // receive new blocks from execution phase
-                    self.process_executed_blocks(blocks, ordered_ledger_info, callback)
-                        .await,
-                    "Error in processing received blocks"
-                );
-            } else {
-                break;
+            select! {
+                CommitChannelType(blocks, ordered_ledger_info, callback) = self.commit_channel_recv.select_next_some() => {
+                    report_err!(
+                        // receive new blocks from execution phase
+                        self.process_executed_blocks(blocks, ordered_ledger_info, callback)
+                            .await,
+                        "Error in processing received blocks"
+                    );
+                }
+                // callback event might come when self.blocks is none
+                reset_event_callback = self.reset_event_rx.select_next_some() => {
```

Comment:
do we need it if the block is already None?

Reply:
Otherwise it will block the caller.